conans/client/cmd/test.py

```diff
@@ -12,31 +12,37 @@ def __init__(self, manager, user_io):
 
     def install_build_and_test(self, conanfile_abs_path, reference, profile,
                                remote, update, build_modes=None, manifest_folder=None,
-                               manifest_verify=False, manifest_interactive=False,):
+                               manifest_verify=False, manifest_interactive=False,
+                               build_folder = None):
         """
         Installs the reference (specified by the parameters or extracted from the test conanfile)
         and builds the test_package/conanfile.py running the test() method.
         """
         base_folder = os.path.dirname(conanfile_abs_path)
-        build_folder = self._build_folder(profile, base_folder)
+        delete_build_folder = build_folder is None
```

Comment:
As you can see Linux tests are passing, Windows tests are failing. This could be due to the fact that Windows hold handles to files open, more time than necessary, so it is not possible to remove the build folder just after building and running the package tests.

Reply:
I already fixed this issue. The problem was that, on windows, deleting the current working directory is not possible.